### PR TITLE
feat: Rich text

### PR DESCRIPTION
**Pull request by ariansobczak-rst**
Resigned from Quill rich text editor and replaced with TipTap (https://tiptap.dev/) 
Introduced a simple menu bar
image
image

### CODE REVIEW

File: entry.js [hunk 1]
```diff
@@ -1,11 +1,9 @@
 import ReactDatepicker from 'react-datepicker'
 import * as CarbonIcons from '@carbon/icons-react'
-import Quill from 'quill'
 
 import * as AdminJSDesignSystem from './src/index'
 
 window.ReactDatepicker = window.ReactDatepicker || ReactDatepicker
 window.CarbonIcons = window.CarbonIcons || CarbonIcons
-window.Quill = Quill
```
Comment (dziraf) [inline entry.js:9]: don't we need the same for tiptap/other editors?

File: package.json [hunk 2]
```diff
@@ -67,12 +66,17 @@
     "typescript": "^3.9.7"
   },
   "dependencies": {
-    "@carbon/icons-react": "^10.14.0",
+    "@carbon/icons-react": "^10.48.0",
+    "@tiptap/extension-document": "^2.0.0-beta.15",
```
Comment (dziraf) [inline package.json:70]: do we have to use beta releases?

File: src/molecules/rich-text-editor/menu-bar.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  name: string
+  onClick: () => void
+  icon?: string
+  attributes?: Record<string, any>
+}
+
+const MenuButton: FC<MenuButtonProps> = (props) => {
+  const { name, editor, onClick, icon, attributes = {} } = props
+
+  const isActive = useCallback(
```
Comment (dziraf) [inline src/molecules/rich-text-editor/menu-bar.tsx:38]: wouldn't `useMemo` be better for this? 🤔

File: src/molecules/rich-text-editor/useTiptapCommands.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+interface TiptapCommand {
+  name: string
+  onClick: () => void
+  icon: string
+  attributes?: Record<string, any>
+}
+
+const useTiptaponClicks = (props: useTiptaponClicksProps): TiptapCommand[] => {
```
Comment (dziraf) [inline src/molecules/rich-text-editor/useTiptapCommands.ts:14]: `useTiptaponClicks` -> `useTiptapOnClicks`? and same for type name

File: src/molecules/rich-text/rich-text.stories.tsx [hunk 5]
```diff
@@ -1,124 +0,0 @@
-/* eslint-disable react/no-danger */
```
Comment (dziraf) [inline src/molecules/rich-text/rich-text.stories.tsx:1]: I think we also need a storybook file for TipTap editor? We'd also need to deploy the updated storybook later

**Comment by dziraf** [comment]
I'd also check if Quill allows to embed images since we'd need the same for Tiptap.

File: entry.js (hunk 1 above)
Comment (ariansobczak-rst) [reply entry.js:9]: I think no, TipTap works without global variables.

File: package.json (hunk 2 above)
Comment (ariansobczak-rst) [reply package.json:70]: Unfortunately, TipTap v1 isn’t supported anymore (https://tiptap.dev/overview/upgrade-guide) and there is no not beta official version https://www.npmjs.com/package/@tiptap/core

File: src/molecules/rich-text-editor/menu-bar.tsx (hunk 3 above)
Comment (ariansobczak-rst) [reply src/molecules/rich-text-editor/menu-bar.tsx:38]: Replaced with useMemo. Thanks!

File: src/molecules/rich-text-editor/useTiptapCommands.ts (hunk 4 above)
Comment (ariansobczak-rst) [reply src/molecules/rich-text-editor/useTiptapCommands.ts:14]: Thanks for catching misspelling, should be `useTiptapCommands`

File: src/molecules/rich-text/rich-text.stories.tsx (hunk 5 above)
Comment (ariansobczak-rst) [reply src/molecules/rich-text/rich-text.stories.tsx:1]: Restored and fit stories for rich text. Fixed stories for currency input.

**Comment by ariansobczak-rst** [comment]
> I'd also check if Quill allows to embed images since we'd need the same for Tiptap.

Added like in Quill base64 representation of an image to handle inline images.

**Review by dziraf — APPROVED** [review]

**Comment by robikovacs** [comment]
Hey @ariansobczak-rst, How do I customize tiptap? Just pass the following to the AdminJS options?

```
custom: {
   extensions:  { ... }
}
```

**Comment by ariansobczak-rst** [comment]
Hi @robikovacs, yes you can pass editor options as a configuration. Unfortunately, I didn't try other image uploads but please don't hesitate to ask in our community https://adminjs.page.link/slack :)

**Comment by benjaminmerchin** [comment]
@robikovacs 
Hello, did you manage to solve this issue, and if you managed, how ? Thanks a lot

**Pull request merged**